plugins/httpapi/nd.py

```diff
@@ -203,6 +210,39 @@ def send_request(self, method, path, data=None):
             raise ConnectionError(json.dumps(self._verify_response(None, method, full_path, None)))
         return self._verify_response(response, method, full_path, rdata)
 
+    def send_file_request(self, method, path, file=None, data=None):
+        ''' This method handles all ND REST API requests other than login '''
+
+        self.error = None
+        self.path = ''
+        self.status = -1
+        self.info = {}
+        self.method = 'GET'
+        if method is not None:
+            self.method = method
+
+        try:
+            # create data field
+            data["uploadedFileName"] = os.path.basename(file)
+            with open('data.json', 'w') as data_file:
+                json.dump(data, data_file)
+        except Exception as e:
+            self.error = dict(code=self.status, message='ND HTTPAPI create data field Exception: {0} '.format(e, traceback.format_exc()))
+            raise ConnectionError(json.dumps(self._verify_response(None, method, path, None)))
+
+        try: 
+            # create fields for MultipartEncoder
+            fields = dict(data=('data.json', open('data.json', 'rb'), 'application/json'), file=(os.path.basename(file), open(file, 'rb'), mimetypes.guess_type(file)))
```

Comment:
Any chance we can move this part of the code back in the NDI side?

Reply:
seems like ansible httpapi plugin doesn't allow us to do that